#!/usr/bin/env python

# set up parameters that we care about
PACKAGE = 'jsk_pcl_ros'

from dynamic_reconfigure.parameter_generator_catkin import *;

from math import pi

gen = ParameterGenerator ()

method_type = gen.enum([gen.const("SAC_RANSAC", int_t, 0, "pcl::SAC_RANSAC"),
                        gen.const("SAC_LMEDS",  int_t, 1, "pcl::SAC_LMEDS"),
                        gen.const("SAC_MSAC",   int_t, 2, "pcl::SAC_MSAC"),
                        gen.const("SAC_RRANSAC",int_t, 3, "pcl::SAC_RRANSAC"),
                        gen.const("SAC_RMSAC",  int_t, 4, "pcl::SAC_RMSAC"),
                        gen.const("SAC_MLESAC", int_t, 5, "pcl::SAC_MLESAC"),
                        gen.const("SAC_PROSAC", int_t, 6, "pcl::SAC_PROSAC"),], "An enum for Segmentation Mehtods")
gen.add("method_type", int_t, 0, "Segmentation Methods", 0, 0, 6, edit_method=method_type)
gen.add("outlier_threshold", double_t, 0, "outlier threshold", 0.005, 0.0, 0.1)
gen.add("max_iterations", int_t, 0, "maximum iteration", 1000, 1, 10000)
gen.add("min_indices", int_t, 0, "maximum iteration", 1000, 1, 10000)
gen.add("min_length", double_t, 0, "minimum length", 0.1, 0, 1.0)
gen.add("line_width", double_t, 0, "line width", 0.01, 0.0, 10000.0)
exit (gen.generate (PACKAGE, "jsk_pcl_ros", "LineSegmentDetector"))
